_Week8/Group3/simulate.do

```diff
@@ -0,0 +1,54 @@
+* first, drop any program called 'schools'
+cap prog drop schools
+
+* Now we create a program 'schools' that will store results in r()
+prog def schools, rclass
+	syntax, rho(real) // input is intracluster correlation coefficient ('ICC')
+	
+	clear // not clear all bc we need to preserve matrices in results.do
+	
+	set obs 200 // 200 clusters (schools)
+	gen school = _n // to refer to each school, which are clusters
+	gen n_students = round(75 + 10*rnormal()) // randomizing number of students (observations) per school (cluster)		obs_per_cluster
+	expand n_students // replace each school obs with "n_students" many copies
+	by school, sort: gen id = _n // assign ids to each student to differentiate them 
+	
+	// generating actual intracluster correlation in the data
+		// assuming total variance = 1
+		local sd_u = sqrt(`rho') // u represents standard dev BETWEEN clusters
+		local sd_e = sqrt(1-`rho') // e is standard dev WITHIN clusters
```

Comment:
This is a very clever and interesting approach, can you tell me more about how you developed it and whether there were alternatives possible?

Reply:
One reason we set total variance equal to 1 is that it would make our program simpler. That is, we could set sd_u = sqrt(rho) and sd_e = sqrt(1-rho).

I suppose one alternative method we could've done would be having TWO inputs to our program: total variance ("term1") and variance between clusters ("term2"), which is the numerator in ICC formula. Then we could set sd_u = sqrt(term2) and sd_e = sqrt(term1-term2).

I'm curious about exactly how this would affect the data we generate, but I think it means we'd have more variability (potentially both between and within clusters). Perhaps I'll look further into this for the final project!

If you see this comment, I'd love to hear your thoughts.